### DISCUSSION TITLE
use NuxtLink & useAsyncData fetch data not refresh

### DISCUSSION THREAD

**Discussion by leosin** [nuxt/nuxt#16083] [General]
I would like to click on the category link to jump to the corresponding category id-slug page
For now, it can be displayed
However, when I click on the category again, nothing happens

I heard that need to use watch & refresh, but how to write it? 
Begging for help 😭

`/news/[id]-[slug].vue`
```vue
<script lang="ts" setup>
const config = useRuntimeConfig()
const route = useRoute()
const params_post = ref({
  categories: useRoute().params.id,
})

const [{ data: categories, refresh }, { data: posts }] = await Promise.all([
  useFetch<any>('/wp/v2/categories', {
    key: 'categories',
    baseURL: config.public.apiBase,
  }),
  useFetch<any>('/wp/v2/posts', {
    key: 'posts',
    baseURL: config.public.apiBase,
    params: params_post,
    // watch: [params_post],
  }),
])
</script>

<template>
  <div>
    <nav class="container">
      <!-- Display Categories -->
      <ul class="flex py-12 justify-between">
        <li v-for="(item, index) in categories" :key="index">
          <nuxt-link :to="{ name: 'demo-nuxt-news-id-slug', params: { id: item.id, slug: item.slug } }" @click="refresh">
            {{ item.id }}-{{ item.name }}
          </nuxt-link>
        </li>
      </ul>
    </nav>

    <div class="container">
      <!-- Display post list -->
      <div v-for="(item, index) in posts" :key="index">
        <h3>{{ item.title.rendered }} <small>id={{ item.categories }}</small></h3>
        <div v-html="item.content.rendered" />
      </div>
    </div>
  </div>
</template>
```
for online editor
https://stackblitz.com/edit/nuxt-starter-h9us1d?file=pages/news/[id]-[slug].vue

**Comment by Flanker72** [comment]
You can see this example in official docs: https://v3.nuxtjs.org/guide/features/data-fetching#refreshing-data

Also, keep in mind, that data caching is performed by provided `key`. Make complex key like `posts-id-${useRoute().params.id}-slug-${useRoute().params.slug}` - and since they will be changed - the data should be refetched.

You can play with such keys or use `refreshNuxtData` to completely refetch anything.

Also you can use watch with useAsyncData - it will refetch related source when watched element is triggered.

And, once more, you may need to make your `params_post` computed instead of ref - as for now, object may not trigger ref update, since it's deep child is updated. Make it computed or reactive - it should help activate watching.

**Comment by leosin** [comment]
thank you for help  @Flanker72 
It turns out that I just need to add `refresh()` on it. LOL
But I don't know if that's the right way to write it

Sorry, I'm really a newbie

```
<script lang="ts" setup>
const config = useRuntimeConfig()
const params_post = {
  categories: useRoute().params.id,
}
const [{ data: categories }, { data: posts, refresh }] = await Promise.all([
  useFetch<any>('/wp/v2/categories', {
    key: 'categories',
    baseURL: config.public.apiBase,
  }),
  useFetch<any>('/wp/v2/posts', {
    key: 'posts',
    baseURL: config.public.apiBase,
    params: params_post,
  }),
])
refresh()
```

**Reply by Flanker72** [reply]
This code will work, however there may be an issue with duplicate queries, especially with SSR. Using `useAsyncData` with watch may help you to avoid calling `refresh()` explicitly.
Initial loading is called automatically within this approach, there's no need to trigger refresh every time by hand.

Data fetching is a really complex task, so you may need to test some different approaches to find the best matching your requirements.

**Comment by leosin** [comment]
I found where the problem is, by default

`initialCache: true`

**Comment by ayitinya** [comment]
I encountered a similar issue and solved it when using a unique key on the useFetch() composable

**Comment by RSS1102** [comment]
I tried to use `refreshNuxtData`,`res.refresh()`,`watch:[pramas]`,`initialCache: false`.They didn't get what I wanted.

and use `refreshNuxtData`,`res.refresh()`,Although he will refresh the page, the page data given will always be the last data.
**my data**: a=>b=>c=>d. **page**:a=>a=>b=>c.

Finally, I set a unique key.and my serevr  set` @ Header ('Access Control Allow Origin', '*') `. I got the expectation.
mention:  https://github.com/nuxt/framework/discussions/5418

**Comment by grafitborz** [comment]
refreshNuxtData works:

const handleClick = () => refreshNuxtData("your-data")
